### PR TITLE
feat(LinkWithIcon): add option for icon on left

### PR DESCRIPTION
**Pull request by kennylam**
### Related Ticket(s)

#2416 

### Description

Adds option to position the icon in `LinkWithIcon` to the left side.

### Changelog

**New**

- `LinkWithIcon` now supports icon positioning on the left side of link

**Changed**

- `TextCTA` now supports icon positioning on the left side of link
- `LinkList` now supports icon positioning on the left side of link for horizontal/vertical layouts

### CODE REVIEW

File: packages/styles/scss/components/link-with-icon/_link-with-icon.scss [hunk 1]
```diff
@@ -47,15 +47,32 @@
       fill: currentColor;
     }
 
-    .#{$prefix}--link--disabled:visited {
-      // Carbon core's has ongoing discussion on disabled link support.
-      // Meanwhile, this CSS makes disabled and visited link makes the right color
-      color: $disabled-02;
+    &.#{$prefix}--link--disabled {
+      &:not([href]),
+      &:not([button]) {
```
Comment (asudoh) [inline packages/styles/scss/components/link-with-icon/_link-with-icon.scss:52]: Question: Wondering what the change is here for. One guess I made is un-disabling link when `href` attribute is there, but `[button]` made me unsure about it. Any insights?

To put this topic in more context; Wondering if it affects CSS overrides scenario like what's described in: https://www.w3.org/TR/selectors-4/#zero-matches

File: packages/react/src/components/LinkWithIcon/LinkWithIcon.js [hunk 2]
```diff
@@ -43,6 +48,11 @@ LinkWithIcon.propTypes = {
    */
   href: PropTypes.string,
 
+  /**
+   * Display icon to left of link.
+   */
+  iconOnLeft: PropTypes.bool,
```
Comment (asudoh) [inline packages/react/src/components/LinkWithIcon/LinkWithIcon.js:54]:
```suggestion
  iconPlacement: PropTypes.oneOf(['left', 'right']),
```

suggestion(non-blocking): Carbon has had to make some breaking changes when we found a need for the 3rd option for where `bool` props are defined.

File: packages/styles/scss/components/link-with-icon/_link-with-icon.scss (hunk 1 above)
Comment (kennylam) [reply packages/styles/scss/components/link-with-icon/_link-with-icon.scss:52]: Thanks @asudoh, I agree. That's in the [current version of Carbon](https://github.com/carbon-design-system/carbon/blob/fd79e780e5c4c49a9a0720490d31cbfa641e8b6e/packages/components/src/components/link/_link.scss#L48-L53) we're using. I planned on removing once #2940 goes through.

Comment (asudoh) [reply packages/styles/scss/components/link-with-icon/_link-with-icon.scss:52]: Makes sense, thank you very much for clarifying @kennylam! (My apologies that I didn’t see it came from Carbon)

File: packages/react/src/components/LinkWithIcon/LinkWithIcon.js (hunk 2 above)
Comment (kennylam) [reply packages/react/src/components/LinkWithIcon/LinkWithIcon.js]: Good call, thanks @asudoh!

**Pull request merged**